Sources/TABResourceLoader/Protocols/Resource types/DataResourceType.swift

```diff
@@ -16,5 +16,5 @@ public protocol DataResourceType: ResourceType {
 
    - returns: An instantiated model if parsing was succesful, otherwise nil
    */
-  func result(from data: Data) -> Result<Model>
+  func result(from data: Data) throws -> Model
```

Comment:
the name `result(from:)` is technically no longer correct. i think it should be `model(from:)`